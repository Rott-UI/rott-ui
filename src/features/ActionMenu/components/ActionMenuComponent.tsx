// React Imports
import React, {FC} from 'react'

// Component Imports
import {Item, Pressable, Label, Button, List, ActionMenu} from '../..'
import {ActionMenuStyles} from '../styles'
import {ActionMenuProps, ActionModel} from '../models'

// Constant Imports

// Util and Lib Imports

/**
 * Action Menu
 *
 * @param title Başlık alanı
 * @param subTitle Alt başlık alanı
 * @param data ActionModel[] formatında veri listesi
 * @param visible ActionMenu komponentin bulunduğu yerdeki Show state i
 * @returns Action Menu Renderlanır
 *
 */
export const ActionMenuComponent: FC<ActionMenuProps> = ({
  title,
  subTitle,
  data,
  itemHeight,
  maxItem,
  separatorTotalHeight,
}) => {
  const itemCount = (data?.length ?? 0) > maxItem! ? maxItem! : data?.length ?? 0

  // Listede renderlanacak Item
  const renderItem = ({
    title: renderItemTitle,
    name: renderItemName,
    action: onAction,
  }: ActionModel) => (
    <Pressable
      testID={`action-menu-data-${renderItemName}-test-id`}
      height={itemHeight}
      justifyContentCenter
      alignItemsCenter
      onPress={() => onAction()}>
      <Label fontSize='lg' variant='grey-900' fontWeight={500}>
        {renderItemTitle}
      </Label>
    </Pressable>
  )

  return (
    <Item size='full' style={ActionMenuStyles().actionMenuContainer} alignItemsCenter>
      <Item
        width={342}
        backgroundColor={COLOURS.GREY100}
        borderTopStartRadius={!(title || subTitle) ? 12 : 0}
        borderTopEndRadius={!(title || subTitle) ? 12 : 0}
        borderBottomStartRadius={12}
        borderBottomEndRadius={12}
        marginBottom={16}>
        <List
          testID='action-menu-test-id'
          style={{minHeight: 2}}
          width={342}
          height={itemHeight! * itemCount + separatorTotalHeight!}
          data={data}
          renderItem={({item}) => renderItem(item)}
          horizontal={false}
          separatorVariant='neutral-alpha-200'
          renderSeparator
          headerSeparator
          footerSeparator
          estimatedItemSize={display.normalize(itemHeight!, 'height')}
          showsVerticalScrollIndicator={false}
          scrollEnabled={(data?.length ?? 0) > maxItem!}
        />
      </Item>

      <Button
        variant='primary'
        onPress={() => ActionMenu.hideActionMenu()}
        fontSize='xl'
        size='full'
        testID='action-menu-cancel-test-id'
        marginBottom={21}>
        {translator('COMMON.CANCEL')}
      </Button>
    </Item>
  )
}
