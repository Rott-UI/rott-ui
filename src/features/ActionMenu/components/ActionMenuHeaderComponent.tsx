// React Imports
import React, {FC} from 'react'

// Util and Lib Imports
import {Item, Label} from '../..'
import {themeConfig} from '../../../providers'

interface ActionMenuHeaderProps {
  title?: Nullable<string>
  subTitle?: Nullable<string>
}

export const ActionMenuHeaderComponent: FC<ActionMenuHeaderProps> = ({title, subTitle}) => {
  return (
    <Item size='full' alignItemsCenter>
      <Item testID='action-menu-header-test-id' width={342}>
        {(title || subTitle) && (
          <Item
            justifyContentCenter
            alignItemsCenter
            size='full'
            backgroundColor={themeConfig.colors['grey-100']}
            borderTopStartRadius={12}
            borderTopEndRadius={12}
            height={54}>
            {title && (
              <Label
                fontSize='xl'
                variant='grey-900'
                fontWeight={700}
                marginBottom={subTitle ? 4 : undefined}>
                {title}
              </Label>
            )}
            {subTitle && (
              <Label fontSize='sm' variant='grey-900' fontWeight={500}>
                {subTitle}
              </Label>
            )}
          </Item>
        )}
      </Item>
    </Item>
  )
}
