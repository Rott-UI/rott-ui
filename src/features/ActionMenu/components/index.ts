export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
