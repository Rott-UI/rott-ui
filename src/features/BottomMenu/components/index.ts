export * from './BottomMenu'
export * from './BottomMenuItem'
