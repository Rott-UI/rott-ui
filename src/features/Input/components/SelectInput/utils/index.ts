export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
