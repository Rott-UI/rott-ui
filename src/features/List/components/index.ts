export * from './List'
export * from './FavoriteSwipeComponent'
