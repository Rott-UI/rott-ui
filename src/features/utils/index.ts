export * from './commonUiTestExtension'
export * from './commonUiStyleProperties'
