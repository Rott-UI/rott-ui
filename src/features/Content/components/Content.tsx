// React Imports
import {FC, PropsWithChildren} from 'react'

// React Native Imports
import {View, ViewProps, StyleSheet, KeyboardAvoidingView, Platform, ScrollView} from 'react-native'

// Feature Imports
import {ContentStyles} from '../styles'

// Model Imports
import {CommonUiProps} from '../../../models'

// Provider Imports
import {themeConfig} from '../../../providers'

// Util and Lib Imports
import {display} from '../../../utils'

interface ContentProps extends PropsWithChildren, CommonUiProps, ViewProps {
  row?: boolean
  noPadding?: boolean
  defaultBackgroundColor?: boolean
  keyboardAvoidingView?: boolean
  keyboardVerticalOffset?: number
  keyboardAvoidingViewContainerPaddingBottom?: number
  scrollEnabled?: boolean
}

export const Content: FC<ContentProps> = ({
  row,
  size,
  noPadding,
  alignItemsCenter,
  justifyContentCenter,
  justifyContentSpaceAround,
  justifyContentSpaceBetween,
  paddingHorizontal = noPadding ? 0 : 24,
  children,
  defaultBackgroundColor,
  backgroundColor,
  style,
  keyboardAvoidingView = false,
  keyboardVerticalOffset = 100,
  keyboardAvoidingViewContainerPaddingBottom = 20,
  scrollEnabled = true,
  ...props
}) => {
  return (
    <View
      style={StyleSheet.flatten([
        ContentStyles({
          size,
          row,
          justifyContentCenter,
          justifyContentSpaceAround,
          justifyContentSpaceBetween,
          alignItemsCenter,
          includeAlignItems: true,
          includeJustifyContent: true,
          paddingHorizontal,
          backgroundColor: defaultBackgroundColor
            ? themeConfig.colors['grey-900']
            : backgroundColor,
          includeBorderRadius: true,
          ...props,
        }).defaultContentStyle,
        style,
      ])}
      {...props}>
      {!keyboardAvoidingView && children}

      {keyboardAvoidingView && (
        <KeyboardAvoidingView
          testID='keyboard-avoiding-view-test-id'
          behavior='padding'
          keyboardVerticalOffset={
            Platform.OS === 'ios' ? display.normalize(keyboardVerticalOffset, 'height') : 0
          }
          enabled>
          <ScrollView
            showsHorizontalScrollIndicator={false}
            showsVerticalScrollIndicator={false}
            contentContainerStyle={{
              paddingBottom: display.px(keyboardAvoidingViewContainerPaddingBottom),
            }}
            scrollEnabled={scrollEnabled}>
            {children}
          </ScrollView>
        </KeyboardAvoidingView>
      )}
    </View>
  )
}
