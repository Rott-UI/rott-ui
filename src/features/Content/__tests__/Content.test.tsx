// Component Imports
import {Content} from '../components'

// Util and Lib Imports
import {render} from '../../../utils'

describe('Content -> Custom Component', () => {
  const testId = {
    keyboardAvoidingViewTestId: 'keyboard-avoiding-view-test-id',
  }

  it('ilk render anında snapshot ile eşleşmeli', () => {
    const rendered = render(<Content />)

    expect(rendered).toMatchSnapshot()
  })

  it('keyboard avoiding view varsayilan ekranda olmamali', () => {
    const {keyboardAvoidingViewTestId} = testId
    const {queryByTestId} = render(<Content />)

    const keyboardAvoidingViewElement = queryByTestId(keyboardAvoidingViewTestId)

    expect(keyboardAvoidingViewElement).not.toBeOnTheScreen()
  })

  it('keyboardAvoidingView verildiginde renderlanmali', () => {
    const {keyboardAvoidingViewTestId} = testId
    const {getByTestId} = render(<Content keyboardAvoidingView />)

    const keyboardAvoidingViewElement = getByTestId(keyboardAvoidingViewTestId)

    expect(keyboardAvoidingViewElement).toBeOnTheScreen()
  })
})
