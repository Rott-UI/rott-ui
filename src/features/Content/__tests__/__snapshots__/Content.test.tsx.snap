// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content -> Custom Component ilk render anında snapshot ile eşleşmeli 1`] = `
<View
  style={
    {
      "alignItems": undefined,
      "backgroundColor": undefined,
      "borderBottomEndRadius": undefined,
      "borderBottomStartRadius": undefined,
      "borderRadius": undefined,
      "borderTopEndRadius": undefined,
      "borderTopStartRadius": undefined,
      "bottom": undefined,
      "flex": undefined,
      "flexDirection": "column",
      "flexGrow": undefined,
      "gap": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "overflow": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": 46,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
      "position": undefined,
      "right": undefined,
      "top": undefined,
      "width": undefined,
      "zIndex": undefined,
    }
  }
/>
`;
